src/fondant/core/schema.py

```diff
@@ -274,6 +274,33 @@ def __repr__(self):
     def __eq__(self, other):
         return vars(self) == vars(other)
 
+    @classmethod
+    def create(  # noqa: PLR0913
```

Comment:
I understand that you want to group this logic somewhere. However now the `Field` class needs to know about all these arguments. I'm not sue this is the best approach.